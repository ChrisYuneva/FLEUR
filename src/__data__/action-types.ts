export const GOODS = {
    REQUEST: 'GOODS_REQUEST',
    SUCCESS: 'GOODS_SUCCESS',
    FAILURE: 'GOODS_FAILURE',
    SET_GENDER: 'GOODS_SET_GENDER',
    SET_TYPE: 'GOODS_SET_TYPE',
};

export const BASKET = {
    REQUEST: 'BASKET_REQUEST',
    SUCCESS: 'BASKET_SUCCESS',
    FAILURE: 'BASKET_FAILURE',
};
