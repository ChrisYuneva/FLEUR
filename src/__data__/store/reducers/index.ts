import { combineReducers } from 'redux';
import goodsReducer, { GoodsState } from './goods';
import basketReducer, { BasketState } from './basket';

export type AppStore = {
    goods: GoodsState;
    basket: BasketState;
};

export default combineReducers<AppStore>({
    goods: goodsReducer,
    basket: basketReducer,
});
