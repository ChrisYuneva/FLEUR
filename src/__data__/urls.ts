import { getNavigations } from '@ijl/cli';

const navigations = getNavigations('repos');

export const baseUrl = navigations['repos'];
export const URLs = {
    basket: {
        url: navigations['link.repos.basket'],
    },
    home: {
        url: navigations['link.repos.home'],
    },
    home_man: {
        url: navigations['link.repos.home_man'],
    },
    showcase_men: {
        url: navigations['link.repos.showcase_men'],
    },
    showcase_women: {
        url: navigations['link.repos.showcase_women'],
    },
    ordering: {
        url: navigations['link.repos.ordering'],
    },
};
