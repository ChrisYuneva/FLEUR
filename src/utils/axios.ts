import axios from 'axios';
import { getConfigValue } from '@ijl/cli';

const baseApiUrl = getConfigValue('repos');
export const commonAxios = axios.create({
    baseURL: baseApiUrl,
    headers: {
        'Content-Type': 'application/json;charset=utf-8',
    },
});
