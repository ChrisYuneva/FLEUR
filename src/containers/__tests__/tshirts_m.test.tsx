import React from 'react';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import mockAdapter from 'axios-mock-adapter';
import axios from 'axios';
import { Provider } from 'react-redux';
import ShowcaseTshirtsM from '../../components/showcase';
import { describe, it, expect } from '@jest/globals';
import { store } from '../../__data__/store';

const multipleRequest = async (mock, responses) => {
    await act(async () => {
        await mock.onAny().reply((config) => {
            const [method, url, params, ...response] = responses.shift();

            if (config.url.includes(url)) {
                return response;
            }
        });
    });
};

describe('Тестирование всего приложения', () => {
    let mock;
    beforeEach(() => {
        mock = new mockAdapter(axios);
    });

    it('Тестируем рендер Showcase', async () => {
        // expect.assertions(2)
        // первый рендер
        const component = mount(
            <Provider store={store}>
                <ShowcaseTshirtsM />
            </Provider>
        );

        expect(component).toMatchSnapshot();

        const response = [
            [
                'POST',
                '/login',
                {},
                200,
                {
                    token: 'xvvxfvxvxfv',
                },
            ],
        ];
        // перехватываем запрос
        await multipleRequest(mock, response);
        component.update();

        expect(component).toMatchSnapshot();
    });
});

// describe('Тестирование', () => {
//
//     it('Тестируем рендер ShowcaseTshirtsM', () => {
//         expect.assertions(1);
//         const component = mount(
//             <Provider store={store}>
//                 <ShowcaseTshirtsM />
//             </Provider>
//         );
//
//         expect(component).toMatchSnapshot();
//     });
// });
