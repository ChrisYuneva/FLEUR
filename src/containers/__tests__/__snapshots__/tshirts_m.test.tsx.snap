// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование всего приложения Тестируем рендер Showcase 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(ShowcaseTshirtsM)>
    <ShowcaseTshirtsM
      addProducts={[Function]}
      gender="FEMALE"
      getProducts={[Function]}
      goodsItems={Array []}
      loading={true}
      type="dress"
    >
      <p>
        Пожалуйста, подождите. Идёт загрузка
      </p>
    </ShowcaseTshirtsM>
  </Connect(ShowcaseTshirtsM)>
</Provider>
`;

exports[`Тестирование всего приложения Тестируем рендер Showcase 2`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(ShowcaseTshirtsM)>
    <ShowcaseTshirtsM
      addProducts={[Function]}
      gender="FEMALE"
      getProducts={[Function]}
      goodsItems={Array []}
      loading={false}
      type="dress"
    >
      <Connect(ShowcaseMen)
        caption="Хорошо скроенные классические футболки — это универсальный предмет гардероба, который незаменим летом, а также может использоваться для создания образов в любое другое время года. Предлагаем мужские футболки всевозможных расцветок и фасонов — выполненные в базовом черном и белом цветах или ярких оттенках."
        title="Футболки"
      >
        <ShowcaseMen
          caption="Хорошо скроенные классические футболки — это универсальный предмет гардероба, который незаменим летом, а также может использоваться для создания образов в любое другое время года. Предлагаем мужские футболки всевозможных расцветок и фасонов — выполненные в базовом черном и белом цветах или ярких оттенках."
          setGender={[Function]}
          setType={[Function]}
          title="Футболки"
        >
          <div
            className="wrapper"
          >
            <nav
              className="nav"
            >
              <Link
                className="name"
                to="/home"
              >
                <a
                  className="name"
                  href="/home"
                >
                  FLEUR
                </a>
              </Link>
              <Link
                className="link"
                to="/showcase_jackets_men"
              >
                <a
                  className="link"
                  href="/showcase_jackets_men"
                >
                  Куртки
                </a>
              </Link>
              <Link
                className="link"
                to="/showcase_shirts_men"
              >
                <a
                  className="link"
                  href="/showcase_shirts_men"
                >
                  Рубашки
                </a>
              </Link>
              <Link
                className="link"
                to="/showcase_hoodies_men"
              >
                <a
                  className="link"
                  href="/showcase_hoodies_men"
                >
                  Толстовки
                </a>
              </Link>
              <Link
                className="link"
                to="/showcase_tshirts_men"
              >
                <a
                  className="link"
                  href="/showcase_tshirts_men"
                >
                  Футболки
                </a>
              </Link>
              <Link
                className="link"
                to="/showcase_jeans_men"
              >
                <a
                  className="link"
                  href="/showcase_jeans_men"
                >
                  Джинсы
                </a>
              </Link>
              <Link
                className="link"
                onClick={[Function]}
                to="/showcase_dress"
              >
                <a
                  className="link"
                  href="/showcase_dress"
                  onClick={[Function]}
                >
                  Женщины
                </a>
              </Link>
            </nav>
            <div
              className="div"
            >
              <header
                className="header"
              >
                <Link
                  className="ref"
                  to="/basket"
                >
                  <a
                    className="ref"
                    href="/basket"
                  >
                    <img
                      className="box"
                      src="FILE_MOCK"
                    />
                    <a
                      className="text"
                    >
                      Корзина
                    </a>
                  </a>
                </Link>
              </header>
              <div
                className="scan"
              >
                <h2>
                  Футболки
                </h2>
                <span
                  className="span"
                >
                  Хорошо скроенные классические футболки — это универсальный предмет гардероба, который незаменим летом, а также может использоваться для создания образов в любое другое время года. Предлагаем мужские футболки всевозможных расцветок и фасонов — выполненные в базовом черном и белом цветах или ярких оттенках.
                </span>
                <div
                  className="show"
                >
                  <div
                    className="showrow"
                  >
                    <MetaTags />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </ShowcaseMen>
      </Connect(ShowcaseMen)>
    </ShowcaseTshirtsM>
  </Connect(ShowcaseTshirtsM)>
</Provider>
`;
